multicodec.go

```diff
@@ -5,11 +5,7 @@ import (
 
 	ipld "github.com/ipld/go-ipld-prime"
 	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
-)
-
-var (
-	_ cidlink.MulticodecDecoder = Decoder
-	_ cidlink.MulticodecEncoder = Encoder
```

Comment:
This check has changed to ipld.Encoder/Decoder in main. That should probably remain.

Reply:
Any particular reason it's needed, though? The init func registration will already fail if either of these breaks the interface.

Reply:
I tend to write these as a form of explicitness and documentation (anytime doing so would not create cycles or dependency sprawl problems).

It's true that there's often other load-bearing code nearby that effectively performs a similar compile time check, but those are often not as clear to a reader who's only browsing the code without active compiler assistance (one would have to look in other files to find the relevant types, etc; and one would have to intuit which functions have such constraints, which is itself entirely nontrivial).

Reply:
Perhaps it's neutral here, since the constraining functions are also right here in the same file, and their purpose is fairly obvious.  I therefore don't have a strongly held opinion about whether the convention is worth it, in this case.

Reply:
I don't have strong opinions either, so I'll undo this.

Reply:
Rebased and reverted this bit.